providers/route53/route53Provider.go

```diff
@@ -12,28 +13,42 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/private/waiter"
 	r53 "github.com/aws/aws-sdk-go/service/route53"
+	r53d "github.com/aws/aws-sdk-go/service/route53domains"
 	"github.com/pkg/errors"
 )
 
 type route53Provider struct {
-	client *r53.Route53
-	zones  map[string]*r53.HostedZone
+	client    *r53.Route53
+	registrar *r53d.Route53Domains
+	zones     map[string]*r53.HostedZone
 }
 
-func newRoute53(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+func newRoute53Reg(conf map[string]string) (providers.Registrar, error) {
+	return newRoute53(conf, nil)
+}
+
+func newRoute53Dsp(conf map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	return newRoute53(conf, metadata)
+}
+
+func newRoute53(m map[string]string, metadata json.RawMessage) (*route53Provider, error) {
 	keyId, secretKey := m["KeyId"], m["SecretKey"]
 
+	// Route53 uses a global endpoint and route53domains
+	// currently only has a single regional endpoint in us-east-1
+	// http://docs.aws.amazon.com/general/latest/gr/rande.html#r53_region
 	config := &aws.Config{
-		Region: aws.String("us-west-2"),
+		Region: aws.String("us-east-1"),
 	}
 
 	if keyId != "" || secretKey != "" {
 		config.Credentials = credentials.NewStaticCredentials(keyId, secretKey, "")
 	}
 	sess := session.New(config)
 
-	api := &route53Provider{client: r53.New(sess)}
+	api := &route53Provider{client: r53.New(sess), registrar: r53d.New(sess)}
```

Comment:
If you aren't using R53 as a registrar, it shouldn't attempt any route53domains actions. You can see on `:52` that it does a `GetZones` to ensure the key is valid, but I didn't add any similar route53domains-y check. So if you try to use it as a registrar without the proper IAM permissions, you'll get an `AccessDeniedException` when it tries to do registrar stuff. But if you don't have r53domains as your registrar, everything should proceed as it has in the past. See below for an example.

If there's a good way to tell ahead of time if the provider is being instantiated as a dsp, a registrar, or both, I'd be happy to add a conditional r53domains access check. Other I'd be willing to break the registrar out into a separate type. But the short answer is that if we do nothing, everything should continue to Just Work for everyone using the r53 dsp.

```bash
# This profile has no IAM permissions to route53domains.
# With route53domains selected as a registrar:
$ ./dnscontrol-Linux preview
Initialized 1 registrars and 1 dns service providers.
******************** Domain: [snip]
----- Getting nameservers from: r53
----- DNS Provider: r53... 1 correction
#1: [snip]

----- Registrar: r53
Error getting corrections: AccessDeniedException: User: arn:aws:iam::868399730840:user/dnscontrol is not authorized to perform: route53domains:GetDomainDetail
	status code: 400, request id: 48b534a1-7902-11e7-afa6-a3fffd2ce139
Done. 1 corrections.

# With none selected as a registrar
$ ./dnscontrol-Linux preview
Initialized 1 registrars and 1 dns service providers.
******************** Domain: [snip]
----- Getting nameservers from: r53
----- DNS Provider: r53... 1 correction
#1: [snip]

----- Registrar: none
Done. 1 corrections.
```